rfcs/0001-restructure.md

```diff
@@ -0,0 +1,285 @@
+
+  This distribution must include popular modules like memcached, mongodb,
+  mysqli, openssl, phpiredis, zlib etc; and a suitable `php.ini`. Downstream
+  buildpacks are expected to obtain info (like extension dir) through the
+  [`php-config`](https://www.php.net/manual/en/install.pecl.php-config.php)
+  executable whenever possible.
+
+* **composer**:
+  Installs [`composer`](https://getcomposer.org), a dependency manager for PHP
+  and makes it available on the `$PATH`
+  * provides: `composer`
+  * requires: none
+
+* **composer-install**:
+  Resolves project dependencies, and installs them using `composer`.
+  * provides: none
+  * requires: `php`, `composer` at `build`
+
+* **php-fpm**:
+  Configures `php-fpm.conf` (config file in `php.ini` syntax), and sets a start
+  command (type `php-fpm`) to start FPM[<sup>1</sup>](#note-1).
+  * provides: `php-fpm`
+  * requires: `php` during launch
+
+  Separation of FPM into a separate buildpack lets users run FPM in one
+  container and web server in another container.
+  The generated `php-fpm.conf` must accept requests on a tcp/ip socket by
+  default, must "include" custom fpm configs specified by the user (e.g. via
+  env variables or specific location in the app).
+
+* **php-builtin-server**:
+  Set up PHP's [built-in web
+  server](https://www.php.net/manual/en/features.commandline.webserver.php) to
+  serve PHP applications.
+  * provides: none
+  * requires: `php` at `launch`
+
+  This buildpack sets a start command (type `web`) to start the built-in web
+  server. This is the default web server.
+
+* **php-httpd**:
+  Sets up HTTPD as the web server to serve PHP applications.
+  * provides: none
+  * requires: `php`, `php-fpm`, `httpd`, at `launch`
+
+  This buildpack generates `httpd.conf` and sets up a start command (type
+  `web`) to run PHP FPM and HTTPD Server. Users need to declare the intention to
+  use httpd.
+
+* **php-nginx**:
+  Sets up Nginx as the web server to serve PHP applications.
+  * provides: none
+  * requires: `php`, `php-fpm`, `nginx` at `launch`
+
+  This buildpack generates `nginx.conf` and sets up a start command (type
+  `web`) to run PHP FPM and Nginx Server. Users need to declare the intention to
+  use nginx.
+
+* **php-memcached-session-handler**:
+  Configures the given memcached service instance as a PHP session
+  handler[<sup>3</sup>](#note-3). Memcached settings are to be provided through
+  a suitable
+  [binding](https://paketo.io/docs/buildpacks/configuration/#bindings).
+  * provides: none
+  * requires: php at `launch`
+
+* **php-redis-session-handler**:
+  Configures the given redis service instance as a PHP session
+  handler[<sup>3</sup>](#note-3). Redis settings are to be provided through a
+  suitable
+  [binding](https://paketo.io/docs/buildpacks/configuration/#bindings).
+  * provides: none
+  * requires: php at `launch`
+
+This would result in the following order groupings in the PHP language family meta-buildpack:
+
+```toml
```

Comment:
@arjun024 - Out of curiosity, is it necessary (or what other spec's have done) to lock the order groups into the spec? It seems like that would lock us into one specific order that we could only change by altering the spec. I'm just a little concerned given we haven't built any of this out yet.

Reply:
@dmikusa-pivotal  This should only serve as a proposed design for guidance. We can incrementally start building up components and swap them into the order. See similar plan for python restructure [here](https://github.com/paketo-community/python/issues/226). In case of a required change, we should be able to deprecate this order and record the new order in another RFC.

Reply:
OK. Just thought I'd mention it. If you're not concerned, neither am I.